src/common/permission/PermissionManager.cpp

```diff
@@ -0,0 +1,214 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "permission/PermissionManager.h"
+
+namespace nebula {
+namespace permission {
+
+// static
+bool PermissionManager::canReadSpace(session::Session *session, GraphSpaceID spaceId) {
+    if (session->isGod()) {
+        return true;
+    }
+    bool havePermission = false;
+    switch (session->roleWithSpace(spaceId)) {
+        case session::Role::GOD :
+        case session::Role::ADMIN :
+        case session::Role::DBA :
+        case session::Role::USER :
+        case session::Role::GUEST : {
+            havePermission = true;
+            break;
+        }
+        case session::Role::INVALID_ROLE : {
+            break;
+        }
+    }
+    return havePermission;
+}
+
+// static
+bool PermissionManager::canReadSchemaData(session::Session *session) {
+    if (session->isGod()) {
```

Comment:
why not put the check after `if (session->space() == -1) `

Reply:
> why not put the check after `if (session->space() == -1)`

Good point.

Reply:
> why not put the check after `if (session->space() == -1)`

fixed.